pkg/asset/installconfig/vsphere/validation.go

```diff
@@ -30,6 +36,35 @@ func ValidateForProvisioning(ic *types.InstallConfig) error {
 	}
 
 	allErrs = append(allErrs, validation.ValidateForProvisioning(ic.Platform.VSphere, field.NewPath("platform").Child("vsphere"))...)
+	allErrs = append(allErrs, folderExists(ic, field.NewPath("platform").Child("vsphere").Child("folder"))...)
 
 	return allErrs.ToAggregate()
 }
+
+// folderExists returns an error if a folder is specified in the vSphere platform but a folder with that name is not found in the datacenter.
+func folderExists(ic *types.InstallConfig, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	cfg := ic.VSphere
+
+	// If no folder is specified, skip this check as the folder will be created.
+	if cfg.Folder == "" {
+		return allErrs
+	}
+
+	vim25Client, _, err := vspheretypes.CreateVSphereClients(context.TODO(), cfg.VCenter, cfg.Username, cfg.Password)
+	if err != nil {
+		err = errors.Wrap(err, "unable to connect to vCenter API")
+		return append(allErrs, field.InternalError(fldPath, err))
+	}
+
+	finder := find.NewFinder(vim25Client)
+
+	path := fmt.Sprintf("/%s/vm/%s", cfg.Datacenter, cfg.Folder)
```

Comment:
what if the user already gave us the folder path prefixed with `/<datacenter>/vm/` ?

Reply:
we should try to error out saying you have prefixed it, we would like un-prefixed.

Reply:
> we should try to error out saying you have prefixed it, we would like un-prefixed.

I can easily add some static validation. 

>what if the user already gave us the folder path prefixed with /<datacenter>/vm/

When `create cluster` is run validation would run to look for a folder called `/<datacenter>/vm//<datacenter>/vm/`. Assuming it doesn't exist, the install would abort.

I have probably been thinking about this too much, but my approach has been to rely on the API validation because it is difficult to determine what is actually a valid name and what isn't.

`/<datacenter>/vm/<datacenter>/vm` is technically a valid folder name. I haven't tested much with the leading `/`... vsphere seems to convert `/` to `%20` or something like that. 

I think we could probably safely statically validate a leading `/` and I can confirm that with a little more testing.

I also think it is a safe assumption that although it might be allowed no one will really name their folder `/<datacenter>/vm/` and it is much more likely they are making a mistake. 

I will work on adding these.

Reply:
@rgolangh suggested that it might be simpler to require the absolute path, which could actually make things simpler and be easier to validate.